docs/guide-disappearance.md

```diff
@@ -3,93 +3,136 @@ id: guide-disappearance
 
-  // wait for appearance
   await waitFor(() => {
-    expect(getByText('the lion king')).toBeInTheDocument()
+    expect(screen.getByText('the lion king')).toBeInTheDocument()
   })
-
-  // wait for appearance and return the element
-  const movie = await findByText('the lion king')
 })
 ```
 
+These do more or less the same thing (`findBy` uses `waitFor` under the hood),
+but the `findBy` version results in simpler code and a better error message.
+
+_Note: `toBeInTheDocument()` comes from [`@testing-library/jest-dom`][jest-dom]_
+
 ## Waiting for disappearance
 
-The `waitForElementToBeRemoved` [async helper][async-api] function uses a
-callback to query for the element on each DOM mutation and resolves to `true`
-when the element is removed.
+When you want to wait until an element has disappeared, you can use
+`waitForElementToBeRemoved` or `waitFor`.
+
+The `waitForElementToBeRemoved` [async wait utility][async-api] takes an element
+as an argument and waits until that element has been removed from the document.
+
+```javascript
+test('movie title no longer present in document', async () => {
+  // element is present
+  const movie = screen.queryByText('the mummy')
+  await waitForElementToBeRemoved(movie)
 
-```jsx
-test('movie title no longer present in DOM', async () => {
-  // element is removed
-  await waitForElementToBeRemoved(() => queryByText('the mummy'))
+  // element has been removed
+  expect(movie).not.toBeInTheDocument()
 })
 ```
 
-Using
+`waitForElementToBeRemoved` works by using a
 [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)
-is more efficient than polling the DOM at regular intervals with `waitFor`.
+internally, which means it responds to the element being removed in a very
+efficient way.
+
+You can also pass an array or a callback to `waitForElementToBeRemoved` -
+[see the documentation for more options](dom-testing-library/api-async.md#waitforelementtoberemoved).
+
+Another option is using the `waitFor` helper.
 
-The `waitFor` [async helper][async-api] function retries until the wrapped function
-stops throwing an error. This can be used to assert that an element disappears
-from the page.
+The `waitFor` [async wait utility][async-api] retries until the wrapped function
+stops throwing an error. Because expectations throw errors when they fail,
+putting one in the wrapped function can be used to wait for the element to be
+removed.
 
-```jsx
+```javascript
 test('movie title goes away', async () => {
   // element is initially present...
   // note use of queryBy instead of getBy to return null
   // instead of throwing in the query itself
   await waitFor(() => {
-    expect(queryByText('i, robot')).not.toBeInTheDocument()
+    expect(screen.queryByText('i, robot')).not.toBeInTheDocument()
   })
 })
 ```
 
+Polling like this is not as efficient as observing for mutations using
+`waitForElementToBeRemoved`, but sometimes it's the best option.
```

Comment:
Are there any good examples of when `waitFor` would work an `waitForElementToBeRemoved` wouldn't in this context? If not, then maybe it shouldn't have equal space on this page.

Reply:
agree 👍 

If not, I wouldn't even mention a "negated" `waitFor` as an alternative to `waitForElementToBeRemoved`

Reply:
I agree, I've removed the `waitFor` option here ✨